### PR TITLE
SOC Flowsheet

### PR DESCRIPTION
**Pull request by dallan-keylogic**
This pull request adds the SOC flowsheet used in recent NMPC and model degradation papers as an example. A Jupyter notebook takes the user through classical control simulations.

----
Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

    I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
    I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

----
📚 Documentation preview 📚: https://idaes-examples--100.org.readthedocs.build/en/100/

### CODE REVIEW

**Comment by lbianchi-lbl** [comment]
To run CI against a version if `idaes-pse` other than `main`, in `requirements-dev.txt`, change

```
idaes-pse @ git+https://github.com/IDAES/idaes-pse@main
```

to

```
idaes-pse @ git+https://github.com/IDAES/idaes-pse@refs/pull/1234/merge
```

where 1234 is the PR number.

File: idaes_examples/mod/power_gen/soc_dynamic_flowsheet.py [hunk 1]
```diff
@@ -0,0 +1,1477 @@
...
+
+        propagate_state(self.feed03)
+        propagate_state(self.sweep03)
+        self.sweep_heater.default_initializer(
+            solver=solver, solver_options=optarg, output_level=outlvl
+        ).initialize(model=self.sweep_heater)
+        self.feed_heater.default_initializer(
+            solver=solver, solver_options=optarg, output_level=outlvl
+        ).initialize(model=self.feed_heater)
+        # self.sweep_heater.initialize(outlvl=outlvl, solver=solver, optarg=optarg)
```
Comment (bpaul4) [inline idaes_examples/mod/power_gen/soc_dynamic_flowsheet.py:996]: Should these be uncommented, or removed?

File: idaes_examples/mod/power_gen/soc_dynamic_flowsheet.py [hunk 2]
```diff
@@ -0,0 +1,1477 @@
...
+        
+        propagate_state(self.hstrm01)
+        self.feed_recycle_split.initialize(outlvl=outlvl, solver=solver, optarg=optarg)
+        propagate_state(self.hstrm02)
+        propagate_state(self.hstrm03)
+
+        self.sweep_exchanger.default_initializer(
+            solver=solver, solver_options=optarg, output_level=outlvl
+        ).initialize(model=self.sweep_exchanger)
+        # self.sweep_exchanger.initialize(outlvl=outlvl, solver=solver, optarg=optarg)
```
Comment (bpaul4) [inline idaes_examples/mod/power_gen/soc_dynamic_flowsheet.py:1037]: Same question for the three commented lines here

**Review by bpaul4 — APPROVED** [review]
All four example files execute with no issues on my local machine, and the diagram looks good. There are a few commented lines that should be uncommented or cleaned up.

File: idaes_examples/mod/power_gen/soc_dynamic_flowsheet.py (hunk 2 above)
Comment (dallan-keylogic) [reply idaes_examples/mod/power_gen/soc_dynamic_flowsheet.py:1037]: Forgot to remove the old versions when updating them to the new initialization format. Will take care of that.

**Pull request merged**